drizzlepac/haputils/catalog_utils.py

```diff
@@ -909,34 +926,54 @@ def identify_sources(self, **pars):
                                                                               reg_rms_median))
                     # Perform manual detection of sources using theoretical PSFs
                     # Initial test data: ictj65
-                    user_peaks = decutils.find_point_sources(self.image.imgname,
-                                                             data=region,
-                                                             box_size=self.param_dict['region_size'],
-                                                             nsigma=self.param_dict['nsigma'],
-                                                             mask=self.image.footprint_mask)
-                    if len(user_peaks) == 0:
+                    try:
+                        user_peaks, source_fwhm = decutils.find_point_sources(self.image.imgname,
+                                                                 data=region,
+                                                                 box_size=self.param_dict['region_size'],
+                                                                 mask=self.image.footprint_mask,
+                                                                 block_size=self.param_dict['block_size'],
+                                                                 diagnostic_mode=self.diagnostic_mode)
+                    except Exception:
+                        # In case we run into out-of-memory error, or any other exception with
+                        # PSF use (like CTE or horribly mismatched PSFs), fail-over to using
+                        # DAOFind mode instead
+                        log.warning("Exception thrown when trying to use PSFs to find sources with UserStarFinder.")
                         user_peaks = None
 
```

Comment:
Should source_fwhm have a default value as it gets used in Line 955?

Reply:
This part of the code (specifically, Lines 929-941) only overrides the default already provided at the start of the method.